### PR TITLE
update pku vae 1.2 training

### PR DESCRIPTION
**Pull request by Songyuanwei**
# What does this PR do?
1. add vae 1.2  training.
2. fix loss scale bug.

Fixes # (issue)

Adds # (feature)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/mindspore-lab/mindone/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the documentation with your changes? E.g. record bug fixes or new features in `What's New`. Here are the
      [documentation guidelines](https://github.com/mindspore-lab/mindcv/wiki/%E6%96%87%E6%A1%A3%E7%BC%96%E5%86%99%E8%A7%84%E8%8C%83)
- [ ] Did you build and run the code without any errors?
- [ ] Did you report the running environment (NPU type/MS version) and performance in the doc? (better record it for data loading, model inference, or training tasks)
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@xxx

### CODE REVIEW

**Review by wtomin — APPROVED** [review]

**Review by SamitHuang — APPROVED** [review]

File: examples/opensora_pku/opensora/models/causalvideovae/model/losses/discriminator.py [hunk 1]
```diff
@@ -13,6 +14,40 @@ def weights_init(m):
...
+            padding=padding,
+            has_bias=has_bias,
+            **kwargs,
+        ).to_float(dtype)
+        self.dtype = dtype
+
+    def construct(self, x):
+        if x.dtype == ms.float32:
+            x = ops.cast(x, self.dtype)
+            return self.conv(x).to(ms.float32)
```
Comment (CaitinZhao) [inline examples/opensora_pku/opensora/models/causalvideovae/model/losses/discriminator.py:46]: Why add the class? Does amp not work?

File: examples/opensora_pku/opensora/models/causalvideovae/model/modules/conv.py [hunk 2]
```diff
@@ -153,4 +154,8 @@ def construct(self, x):
             # first_frame_pad = mint.repeat_interleave([first_frame], self.time_kernel_size - 1, 2)
             x = ops.concat((first_frame_pad, x), axis=2)
 
-        return self.conv(x)
+        if x.dtype == ms.float32:
```
Comment (CaitinZhao) [inline examples/opensora_pku/opensora/models/causalvideovae/model/modules/conv.py:157]: same as above

File: examples/opensora_pku/scripts/causalvae/train_with_gan_loss.sh [hunk 3]
```diff
@@ -1,12 +1,12 @@
 python opensora/train/train_causalvae.py \
-    --exp_name "25x256x256" \
+    --exp_name "9x256x256" \
     --train_batch_size 1 \
-    --precision bf16 \
+    --precision fp32 \
```
Comment (CaitinZhao) [inline examples/opensora_pku/scripts/causalvae/train_with_gan_loss.sh:4]: why use fp32？

Comment (Songyuanwei) [reply examples/opensora_pku/scripts/causalvae/train_with_gan_loss.sh:4]: bf16混合精度训练 loss不收敛，使用fp32后正常

File: examples/opensora_pku/opensora/models/causalvideovae/model/losses/discriminator.py (hunk 1 above)
Comment (Songyuanwei) [reply examples/opensora_pku/opensora/models/causalvideovae/model/losses/discriminator.py:46]: bf16混合精度训练loss不收敛，需要使用fp32训练，conv3d算子不支持fp32，需要cast

**Review by CaitinZhao — APPROVED** [review]

**Pull request merged**